ygot/diff_test.go

```diff
@@ -249,6 +251,130 @@ func TestNodeValuePath(t *testing.T) {
 		},
 		inSchemaPaths: [][]string{{"foo", "bar"}, {"baz"}},
 		wantErr:       "could not find path specification annotation",
+	}, {
+		desc: "nodeinfo for list member",
+		inNI: &util.NodeInfo{
+			Parent: &util.NodeInfo{
+				Annotation: []interface{}{&pathSpec{
+					gNMIPaths: []*gnmipb.Path{{
+						Elem: []*gnmipb.PathElem{{
+							Name: "a-list",
+						}},
+					}},
+				}},
+			},
+			FieldValue: reflect.ValueOf(&basicListMember{ListKey: String("key-value")}),
+		},
+		wantPathSpec: &pathSpec{
+			gNMIPaths: []*gnmipb.Path{{
+				Elem: []*gnmipb.PathElem{{Name: "a-list", Key: map[string]string{"list-key": "key-value"}}},
+			}},
+		},
+	}, {
+		desc: "nodeinfo for invalid list member",
+		inNI: &util.NodeInfo{
+			Parent: &util.NodeInfo{
+				Annotation: []interface{}{&pathSpec{
+					gNMIPaths: []*gnmipb.Path{{
+						Elem: []*gnmipb.PathElem{{
+							Name: "a-list",
+						}},
+					}},
+				}},
+			},
+			FieldValue: reflect.ValueOf(&errorListMember{StringValue: String("foo")}),
+		},
+		wantErr: "invalid key map",
+	}, {
+		desc: "nodeinfo for list member with unstringable key",
+		inNI: &util.NodeInfo{
+			Parent: &util.NodeInfo{
+				Annotation: []interface{}{&pathSpec{
+					gNMIPaths: []*gnmipb.Path{{
+						Elem: []*gnmipb.PathElem{{
+							Name: "a-list",
+						}},
+					}},
+				}},
+			},
+			FieldValue: reflect.ValueOf(&badListKeyType{Value: &cmplx}),
+		},
+		wantErr: "cannot convert keys to map[string]string: cannot convert type complex128 to a string for use in a key: (1+2i)",
```

Comment:
(optional) Back to our discussion about matching on error messages - consider at least making the matched error shorter and use a substring match. In this case this test actually verifies the error content that is produced by a function different than the one under test. Maybe just matching on "cannot convert keys" would be enough here and be less "change-detectory" at the same time?

Reply:
SGTM -- made this a substring match of the relevant error from this function.

Thinking about it, I wonder whether using something like https://godoc.org/google.golang.org/grpc/status could help here -- since this gives a structured code/error. I'll investigate doing this going forward.